processing/src/main/java/io/druid/query/aggregation/post/MathPostAggregator.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.post;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Sets;
+import io.druid.math.expr.Expr;
+import io.druid.math.expr.Parser;
+import io.druid.query.aggregation.PostAggregator;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ */
+public class MathPostAggregator implements PostAggregator
+{
+  private static final Comparator<Number> DEFAULT_COMPARATOR = new Comparator<Number>()
+  {
+    @Override
+    public int compare(Number o1, Number o2)
+    {
+      if (o1 instanceof Long && o2 instanceof Long) {
+        return Long.compare(o1.longValue(), o2.longValue());
+      }
+      return Double.compare(o1.doubleValue(), o2.doubleValue());
+    }
+  };
+
+  private final String name;
+  private final String expression;
+  private final Comparator comparator;
+  private final String ordering;
+
+  private final Expr parsed;
+  private final List<String> dependentFields;
+
+  public MathPostAggregator(
+      String name,
+      String fnName
+  )
+  {
+    this(name, fnName, null);
+  }
+
+  @JsonCreator
+  public MathPostAggregator(
+      @JsonProperty("name") String name,
+      @JsonProperty("expression") String expression,
+      @JsonProperty("ordering") String ordering
+  )
+  {
+    Preconditions.checkArgument(expression != null, "expression cannot not be null");
+
+    this.name = name;
+    this.expression = expression;
+    this.ordering = ordering;
+    this.comparator = ordering == null ? DEFAULT_COMPARATOR : Ordering.valueOf(ordering);
+
+    this.parsed = Parser.parse(expression);
+    this.dependentFields = Parser.findRequiredBindings(parsed);
```

Comment:
Why not store this as a Set? it seems we only use it for `getDependentFields` anyway.